### PR TITLE
Statedump notifier

### PR DESCRIPTION
**Pull request by tahini**
Those are the tests for the statedump notifier UST feature, available in the following pull request: https://github.com/lttng/lttng-ust/pull/53

### CODE REVIEW

**Comment by tahini** [comment]
I guess this feature will not be reviewed until proper documentation is provided, right? I'll do that in a while.

**Comment by tahini** [comment]
Documentation added in the lttng-ust PR

File: tests/regression/ust/statedump-notifier/test_statedump_notifier [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+	local file_termination=$(mktemp -u)
+	run_app_bg ${file_termination}
+
+	start_lttng_sessiond
+
+	create_lttng_session_ok $SESSION_NAME $TRACE_PATH
+	enable_ust_lttng_event_ok $SESSION_NAME $EVENT_NAME
+	start_lttng_tracing_ok $SESSION_NAME
+
+#	while [ ! -f "${STATEDUMP_FILE}" ]; do
```
Comment (jgalar) [inline tests/regression/ust/statedump-notifier/test_statedump_notifier]: Should this be uncommented?

File: tests/utils/testapp/gen-ust-statedump-events/gen-ust-statedump-events.c [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+static struct lttng_ust_notifier notifier;
+
+static void notifier_cb(struct lttng_session *session, void *priv)
+{
+	/* This file synchronizes with test control to make sure the statedump ran */
+	char *statedump_done_file = (char *) priv;
+	tracepoint(tp, tpteststdmp, session, 123, "test string");
+	tracepoint(tp, tpteststdmp, session, 444, "another string");
+	if (statedump_done_file) {
+		// Somebody may be waiting on this thread, how to tell it if file creation went wrong?
```
Comment (jgalar) [inline tests/utils/testapp/gen-ust-statedump-events/gen-ust-statedump-events.c]: Please use `/* ... */`-style comments rather than `// ...`

**Comment by tahini** [comment]
I'll open a new PR for this one

**Pull request closed without merging**